src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1493,10 +1493,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Dim codeGen = New codeGen.CodeGenerator(method, block, builder, moduleBuilder, diagnostics, optimizations, generateDebugInfo)
 
                 ' We need to save additional debugging information for MoveNext of an async state machine.
-                Dim stateMachineMethod = TryCast(method, SynthesizedStateMachineMethod)
+                Dim stateMachineMoveNextMethod = TryCast(method, SynthesizedStateMachineMoveNextMethod)
```

Comment:
I'm not as concerned about this pattern in the VB compiler as it is already very widely used.